### DISCUSSION TITLE
Modelling of torsion spring

### DISCUSSION THREAD

**Discussion by jgueldenstein** [bulletphysics/bullet3#3788] [General]
I would like to model a Robot with an actuator and a torsion spring after the actuator in the kinematic chain. I have looked at #2100 but i can not figure out how to not allow translation in all axes and rotation around two axes or how to modify a fixed joint from a URDF to achieve this behavior.

My setup can be found here: [spring_test.zip](https://github.com/bulletphysics/bullet3/files/4724363/spring_test.zip)

Thank you in advance.

**Comment by nyxaria** [comment]
Did you make any progress on this? I am trying to model a compression spring myself, as a prismatic joint

**Comment by jgueldenstein** [comment]
Unfortunately I did not make any progress so far
